packages/webpack/test/HoneybadgerSourceMapPlugin.test.js

```diff
@@ -1,533 +1,257 @@
       expect(compilation.warnings.length).to.eq(0)
     })
 
+    it('should call sendDeployNotification if deploy is true', async () => {
+      plugin.options.deploy = true
+      sinon.stub(plugin, 'uploadSourceMaps')
+      sinon.stub(plugin, 'sendDeployNotification')
+
+      await plugin.afterEmit(compilation)
+      expect(plugin.sendDeployNotification.callCount).to.eq(1)
+      expect(plugin.sendDeployNotification.calledWith(plugin.options)).to.equal(true)
+    })
+
+    it('should call sendDeployNotification if deploy is a deploy object', async () => {
+      plugin.options.deploy = {
+        environment: 'test',
+        repository: 'https://cdn.example.com',
+        localUsername: 'itsMeHi'
+      }
+      sinon.stub(plugin, 'uploadSourceMaps')
+      sinon.stub(plugin, 'sendDeployNotification')
+      
+      await plugin.afterEmit(compilation)
+      expect(plugin.sendDeployNotification.callCount).to.eq(1)
+      expect(plugin.sendDeployNotification.calledWith(plugin.options)).to.equal(true)
+    })
+
     it('should add upload warnings to compilation warnings, ' +
       'if ignoreErrors is true and silent is false', async function () {
-      const compilation = {
-        errors: [],
-        warnings: []
-      }
-      this.plugin.ignoreErrors = true
-      this.plugin.silent = false
+      plugin.options.ignoreErrors = true
+      plugin.options.silent = false
 
-      sinon.stub(this.plugin, 'uploadSourceMaps')
+      sinon.stub(plugin, 'uploadSourceMaps')
         .callsFake(() => { throw new Error() })
 
-      await this.plugin.afterEmit(compilation)
+      await plugin.afterEmit(compilation)
 
-      expect(this.plugin.uploadSourceMaps.callCount).to.eq(1)
+      expect(plugin.uploadSourceMaps.callCount).to.eq(1)
       expect(compilation.errors.length).to.eq(0)
       expect(compilation.warnings.length).to.eq(1)
       expect(compilation.warnings[0]).to.be.an.instanceof(Error)
     })
 
     it('should not add upload errors to compilation warnings if silent is true', async function () {
-      const compilation = {
-        errors: [],
-        warnings: []
-      }
-      this.plugin.ignoreErrors = true
-      this.plugin.silent = true
+      plugin.options.ignoreErrors = true
+      plugin.options.silent = true
 
-      sinon.stub(this.plugin, 'uploadSourceMaps')
+      sinon.stub(plugin, 'uploadSourceMaps')
         .callsFake(() => { throw new Error() })
 
-      await this.plugin.afterEmit(compilation)
+      await plugin.afterEmit(compilation)
 
-      expect(this.plugin.uploadSourceMaps.callCount).to.eq(1)
+      expect(plugin.uploadSourceMaps.callCount).to.eq(1)
       expect(compilation.errors.length).to.eq(0)
       expect(compilation.warnings.length).to.eq(0)
     })
 
     it('should add upload errors to compilation errors', async function () {
-      const compilation = {
-        errors: [],
-        warnings: []
-      }
-      this.plugin.ignoreErrors = false
+      plugin.options.ignoreErrors = false
```

Comment:
I think it would be good to assert that there was no attempt to send a deploy notification here (or it could be a test on its own if you prefer - "does not send deploy notification when compilation fails".